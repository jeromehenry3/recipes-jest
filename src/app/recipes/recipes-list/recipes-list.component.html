<div class="row">
  recipes list
</div>
