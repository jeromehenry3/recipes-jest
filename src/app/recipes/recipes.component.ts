import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-recipes',
  templateUrl: './recipes.component.html',
  styleUrls: ['./recipes.component.scss']
})
export class RecipesComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

  fakeFuncTocheckCoverageIsOk() {
    let a = 1;
    a = 2;
    const c = 3;
    console.log('c', c);
    return a + c;
  }

}
