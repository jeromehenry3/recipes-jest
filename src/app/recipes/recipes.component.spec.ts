import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MockComponent } from 'ng-mocks';

import { RecipesComponent } from './recipes.component';
import { RecipeDetailComponent } from './recipe-detail/recipe-detail.component';
import { RecipesListComponent } from './recipes-list/recipes-list.component';
import { By } from '@angular/platform-browser';

describe('RecipesComponent', () => {
  let component: RecipesComponent;
  let fixture: ComponentFixture<RecipesComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        RecipesComponent,
        MockComponent(RecipesListComponent),
        MockComponent(RecipeDetailComponent),
      ],
      providers: []
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(RecipesComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    const recipes = fixture.debugElement.nativeElement;
    expect(recipes).toMatchSnapshot();
  });

  describe('renders child components',() => {
    it('should render `RecipeListComponent`', () => {
      const recipesList = fixture.debugElement.query(By.directive(RecipesListComponent));
      expect(recipesList.componentInstance).toBeTruthy();
    });
    it('should render `RecipeDetailsComponent`', () => {
      const recipeDetails = fixture.debugElement.query(By.directive(RecipeDetailComponent));
      expect(recipeDetails.componentInstance).toBeTruthy();
    });
  });
});
