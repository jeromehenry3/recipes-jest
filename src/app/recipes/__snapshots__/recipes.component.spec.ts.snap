// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="row"
  >
    <div
      class="col-md-5"
    >
      <app-recipes-list />
    </div>
    <div
      class="col-md-7"
    >
      <app-recipe-detail />
    </div>
  </div>
</div>
`;
